libpromises/unix.c

```diff
@@ -202,6 +203,19 @@ bool ShellCommandReturnsZero(const char *command, ShellType shell)
         else
         {
             char **argv = ArgSplitCommand(command);
+            int devnull;
+
+            if ((devnull = safe_open("/dev/null", O_WRONLY)) == -1)
```

Comment:
Another detail: This is not Windows safe. Put a constant in ``platform.h``, for example ``PLATFORM_NULL``, make it ``nul`` for ``#ifdef _WIN32`` and ``/dev/null`` for all others, and then use the constant here.